packages/workbox-google-analytics/test/sw/index.js

```diff
@@ -0,0 +1,349 @@
+/*
+ Copyright 2017 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import IDBHelper from '../../../../lib/idb-helper.js';
+import * as googleAnalytics from '../../src/index.js';
+import constants from '../../src/lib/constants.js';
+import {QueuePlugin} from '../../../workbox-background-sync/src/index.js';
+import {NetworkFirst, NetworkOnly}
+    from '../../../workbox-runtime-caching/src/index.js';
+
+const PAYLOAD = 'v=1&t=pageview&tid=UA-12345-1&cid=1&dp=%2F';
+
+const sleep = (time) => new Promise((resolve) => setTimeout(resolve, time));
+
+const waitUntil = async (test) => {
+  if (test() === true) {
+    return Promise.resolve();
+  } else {
+    await sleep(100);
+    return waitUntil(test);
+  }
+};
+
+describe(`initialize`, function() {
+  let idbMap = {};
+
+  const stubIdbHelper = () => {
+    sinon.stub(IDBHelper.prototype, 'get').callsFake(function(key) {
+      const dbStore = `${this._name}|${this._storeName}`;
+      idbMap[dbStore] = idbMap[dbStore] || {};
+      return Promise.resolve(idbMap[dbStore][key]);
+    });
+    sinon.stub(IDBHelper.prototype, 'put').callsFake(function(key, value) {
+      const dbStore = `${this._name}|${this._storeName}`;
+      idbMap[dbStore] = idbMap[dbStore] || {};
+      idbMap[dbStore][key] = value;
+      return Promise.resolve();
+    });
+  };
+
+  const restoreIdbHelper = () => {
+    idbMap = {};
+    IDBHelper.prototype.get.restore();
+    IDBHelper.prototype.put.restore();
+  };
+
+  const trackAddedEventListeners = () => {
+    sinon.spy(self, 'addEventListener');
+  };
+
+  const removeAllEventListeners = () => {
+    for (const [evt, listener] of self.addEventListener.args) {
+      self.removeEventListener(evt, listener);
+    }
+    self.addEventListener.restore();
+  };
+
+  beforeEach(function() {
+    trackAddedEventListeners();
+    stubIdbHelper();
+  });
+
+  afterEach(function() {
+    removeAllEventListeners();
+    restoreIdbHelper();
+  });
+
+  it(`should register a handler to cache the analytics.js script`, function() {
```

Comment:
This test could have been a browser level test (i.e.  register a sw that just init's the analytics library, requests analytics JS and then checks the cache).